src/methods/wallet-get-calls-status.ts

```diff
@@ -0,0 +1,85 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { Infer } from '@metamask/superstruct';
+import {
+  nonempty,
+  optional,
+  mask,
+  string,
+  array,
+  object,
+  tuple,
+} from '@metamask/superstruct';
+import type {
+  Json,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from '@metamask/utils';
+import { HexChecksumAddressStruct, StrictHexStruct } from '@metamask/utils';
+
+import { validateParams } from '../utils/validation';
+
+const GetCallsStatusStruct = tuple([nonempty(string())]);
+
+const GetCallsStatusReceiptStruct = object({
+  logs: array(
+    object({
+      address: HexChecksumAddressStruct,
+      data: StrictHexStruct,
+      topics: array(StrictHexStruct),
+    }),
+  ),
+  status: StrictHexStruct,
+  chainId: optional(StrictHexStruct),
```

Comment:
is chainId optional in the result? I know we mean for it to be optional in the params of these new methods

Reply:
This struct is only a means to remove any excess properties.

We don't want to validate the wallet itself so I've made all the properties optional, though I'll follow up to confirm why `chainId` was missing in my tests.